troubleshooting/checkpointing/src/solution/java/com/ververica/flink/training/solutions/CheckpointingJobSolution2.java

```diff
@@ -169,11 +169,14 @@ public void process(
 
             long eventsPerWindow = 0L;
             double sumPerWindow = 0.0;
+
+            double diffPerMeasurement;
             Measurement previous = null;
             for (Tuple2<Measurement, Long> measurement : list) {
                 ++eventsPerWindow;
                 if (previous != null) {
-                    sumPerWindow += measurement.f0.getValue() - previous.getValue();
+                    diffPerMeasurement = measurement.f0.getValue() - previous.getValue();
```

Comment:
please define `diffPerMeasurement` here, not outside the loop

Reply:
Done too